test/e2e/utils.go

```diff
@@ -311,6 +311,16 @@ func buildContainer(name, image string, cmd []string) corev1.Container {
 		Name:    name,
 		Image:   image,
 		Command: cmd,
+		SecurityContext: &corev1.SecurityContext{
+			AllowPrivilegeEscalation: pointer.Bool(false),
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{"ALL"},
+			},
+			RunAsNonRoot: pointer.Bool(false),
```

Comment:
This is a bit of a nit, but `buildContainer` appears to be intended to be for generic container object creation for all e2e tests, but we have `RunAsNonRoot: pointer.Bool(false),` which appears to be a bit specific to the container itself (specifically the DNS E2E tests that use this function).

Maybe all E2E tests need to run as root? Does it make sense to parameterize this `RunAsNonRoot` value?

Reply:
That's a good catch and actually looks like the source of a failure: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-dns-operator/339/pull-ci-openshift-cluster-dns-operator-master-e2e-aws-operator/1565792699041189888

I'll add the parameter in so it can be adjusted per use case.

Reply:
Oh actually, it looks like there are 3 occurrences of `buildPod()` which uses `buildContainer()` under the covers. DNS Forwarding, DNS Logging, and DNS-over-TLS tests use it but where they're all using the cli image they have to have root in the container. I think the short term fix is probably to run the test clients in a privileged namespace and mid to long term we should think about using a non-privileged container image. With that said, I'll move the client for DNS logging to `openshift-dns`